native/src/main/java/io/ballerina/stdlib/email/server/EmailListener.java

```diff
@@ -61,15 +62,17 @@ public EmailListener(Runtime runtime) {
      * @return If successful return true
      */
     public boolean onMessage(EmailEvent emailEvent) {
-        Object email = emailEvent.getEmailObject();
-        if (runtime != null) {
-            Set<Map.Entry<String, BObject>> services = registeredServices.entrySet();
-            for (Map.Entry<String, BObject> service : services) {
-                runtime.callMethod(service.getValue(), ON_MESSAGE, null, email);
+        Thread.startVirtualThread(() -> {
+            Object email = emailEvent.getEmailObject();
+            if (runtime != null) {
+                Set<Map.Entry<String, BObject>> services = registeredServices.entrySet();
+                for (Map.Entry<String, BObject> service : services) {
+                    runtime.callMethod(service.getValue(), ON_MESSAGE, new StrandMetadata(true, null), email);
```

Comment:
We should check isolated logic here instead of passing true

Reply:
Fixed in ab350011f8f78f36a5f9f8243243e8f7b0f8f4e4

Reply:
Fixed in 66edec95f5e3194721b310841be7b9ed951414ff